galite-core/src/main/kotlin/org/kopi/galite/report/VReport.kt

```diff
@@ -15,6 +15,641 @@
+    if (enable) {
+      activeCommands.addElement(command)
+    } else {
+      activeCommands.removeElement(command)
+    }
+  }
+
+  override fun createPrintJob(): PrintJob {
+    val exporter: PExport2PDF
+    val printJob: PrintJob
+    exporter = PExport2PDF((getDisplay() as UReport).table,
+            model,
+            printOptions,
+            pageTitle,
+            firstPageHeader,
+            Message.getMessage("toner_save_mode") == "true")
+    printJob = exporter.export()
+    printJob.setDocumentType(documentType)
+    return printJob
+  }
+  /**
+   * Prints the report
+   */
+  /**
+   * Prints the report
+   */
+  fun export(type: Int = TYP_CSV) {
+    val ext: String = when (type) {
+      TYP_CSV -> ".csv"
+      TYP_PDF -> ".pdf"
+      TYP_XLS -> ".xls"
+      TYP_XLSX -> ".xlsx"
+      else -> throw InconsistencyException("Export type unkown")
+    }
+    val file: File = FileHandler.fileHandler.chooseFile(getDisplay(),
+            ApplicationConfiguration.getConfiguration().getDefaultDirectory(),
+            "report$ext")
+    file?.let { export(it, type) }
+  }
+
+  /**
+   * Prints the report
+   */
+  fun export(file: File, type: Int = TYP_CSV) {
+    setWaitInfo(VlibProperties.getString("export-message"))
+    val extension: String
+    val exporter: PExport
+    when (type) {
+      TYP_CSV -> {
+        extension = ".csv"
+        exporter = PExport2CSV((getDisplay() as UReport).table,
+                model,
+                printOptions,
+                pageTitle)
+      }
+      TYP_PDF -> {
+        extension = ".pdf"
+        exporter = PExport2PDF((getDisplay() as UReport).table,
+                model,
+                printOptions,
+                pageTitle,
+                firstPageHeader,
+                Message.getMessage("toner_save_mode") == "true")
+      }
+      TYP_XLS -> {
+        extension = ".xls"
+        exporter = PExport2XLS((getDisplay() as UReport).table,
+                model,
+                printOptions,
+                pageTitle)
+      }
+      TYP_XLSX -> {
+        extension = ".xlsx"
+        exporter = PExport2XLSX((getDisplay() as UReport).table,
+                model,
+                printOptions,
+                pageTitle)
+      }
+      else -> throw InconsistencyException("Export type unkown")
+    }
+    exporter.export(file)
+    unsetWaitInfo()
+    fireFileProduced(file, pageTitle + "_" + Date.now().format("yyyyMMdd") + extension)
+  }
+
+  /**
+   * Sets the title
+   */
+  fun setPageTitle(title: String) {
+    pageTitle = title
+  }
+
+  fun setPageTitleParams(param: Any) {
+    setPageTitleParams(arrayOf(param))
+  }
+
+  fun setPageTitleParams(param1: Any, param2: Any) {
+    setPageTitleParams(arrayOf(param1, param2))
+  }
+
+  fun setPageTitleParams(params: Array<Any>) {
+    setPageTitle(MessageFormat.format(pageTitle, *params))
+  }
+
+  fun setFirstPageHeader(firstPageHeader: String) {
+    this.firstPageHeader = firstPageHeader
+  }
```

Comment:
No , it's better to use "this." when referencing with same variable name as property name , to avoid variable shadowing  and better for code readability .